### DISCUSSION TITLE
Reproducing Rise of AI over 8 decades plot {plotnine} vs {ggplot2}

### DISCUSSION THREAD

**Discussion by Gchism94** [has2k1/plotnine#827] [2024 Plotnine Contest]
### Authors

Greg Chism

### Links

[github](https://github.com/Gchism94), [website](https://gregtchism.com/), [linkedin](https://www.linkedin.com/in/greg-chism/)

### Full description

**Project link**: https://gchism.quarto.pub/plotnineriseofai/

This project uses {plotnine} to reproduce my {ggplot2} version of [The rise of artificial intelligence over the last 8 decades](https://ourworldindata.org/brief-history-of-ai) ([Roser 2022](https://gchism.quarto.pub/plotnineriseofai/#ref-owid-brief-history-of-ai)) plot by [Our World in Data](https://ourworldindata.org/). The plot has some slight differences based on the data used in my version vs the original.

Overall, I found that {plotnine} is not a perfect facsimile of {ggplot2} due to it being built upon the {matplotlib} framework, but its pretty close:

Most of the functionality from my {ggplot2} version was translatable, with some exceptions ([see challenges section](https://github.com/has2k1/plotnine/discussions/new?category=2024-plotnine-contest#challenges)).

The "heart" of {ggplot2} seems to be in {plotnine}, making it a good way for R users to get into plotting through Python.

All-in-all, I think the plot I produced was pretty similar, so clearly the {plotnine} developers did a great job with the package.

This all being said, I found that working with {plotnine} felt like I was going to break it with every new line. The package feels a little fragile due to this. Some of this however is from my lack of experience in Python.

---

#### Disclosure

The original plot is licensed under [CC-BY](https://creativecommons.org/licenses/by/4.0/) by the authors Charlie Giattino, Edouard Mathieu, and Max Roser. This project is licensed under the same.

The data is taken from ([Sevilla et al. 2022](https://gchism.quarto.pub/plotnineriseofai/#ref-sevilla2022)) – [Compute Trends Across Three eras of Machine Learning](https://arxiv.org/pdf/2202.05924.pdf). Published in arXiv on March 9, 2022. The data are [freely available](https://docs.google.com/spreadsheets/d/1AAIebjNsnJj_uKALHbXNfn3_YsT6sHXtCU0q7OIPuc4/edit#gid=0) to the public. Note that the available data to the public continues to be updated.

The original plot has fewer data points, likely due to additions by the data authors after the original article’s release.

### Code repository

**GitHub Repo**: https://github.com/Gchism94/riseOfAIPlot/tree/main

**Quarto Doc**: https://github.com/Gchism94/riseOfAIPlot/blob/main/plotnineRiseOfAI.qmd

_Note_: My original version is also in the GitHub repo, which only uses {ggplot2}, please disregard (named `riseOfAIPlot.qmd`)

**Comment by has2k1** [comment]
![roai](https://github.com/user-attachments/assets/0cc821e2-d216-43fe-be41-4ce7d249cc0d)

**Comment by EconomiCurtis** [comment]
Ahoy @Gchism94 , 
I am one of the contest organizers. 
If you'd like, we'd be happy to send you a little Plotnine and Posit care package of hex-shaped stickers. Just a small thanks for taking part. https://forms.gle/M8QmM286MSHbkDRd7

**Comment by Gchism94** [comment]
@EconomiCurtis, sounds great, thank you!

**Comment by has2k1** [comment]
@Gchism94, The winning submission has been announced [here](https://posit.co/blog/winner-of-the-2024-plotnine-plotting-contest/). Thank you for taking part in the contest.
